test/check-machines-networks

```diff
@@ -795,6 +796,75 @@ class TestMachinesNetworks(VirtualMachinesCase):
         self.waitNetworkRow("test_network2", connectionName, False)
 
     def testNetworkAddStaticDCHPHosts(self):
+        class NetworkAddStaticDCHPHostsDialog(object):
+            def __init__(
+                self, test_obj=None, network_name="default", connection_name="system", mac=None, ip=None, entry_id=0,
+                xfail=None, remove=True
+            ):
+                self.test_obj = test_obj
+                self.network_name = network_name
+                self.connection_name = connection_name
+                self.mac = mac
+                self.ip = ip
+                self.entry_id = entry_id
+                self.xfail = xfail
+                self.remove = remove
+
+            def execute(self):
+                self.open()
+                self.fill()
+                self.add()
+                if not self.xfail:
+                    self.verify_dialog()
+                    self.verify_overview()
+                    if self.remove:
+                        self.cleanup()
+
+            def open(self):
+                b.click(f"#network-{self.network_name}-{self.connection_name}-static-host-entries-add")
+                b.wait_visible("#add-new-static-entry-mac-address")
+                b.wait_in_text(".pf-c-modal-box .pf-c-modal-box__header .pf-c-modal-box__title", "Add a DHCP static host entry")
+
+            def fill(self):
+                b.set_input_text("#add-new-static-entry-mac-address", self.mac)
+                b.set_input_text("#add-new-static-entry-ip-address", self.ip)
+
+            def cancel(self):
+                b.click(".pf-c-modal-box__footer button:contains(Cancel)")
+                b.wait_not_present("#add-new-static-entry")
+
+            def add(self):
+                b.click(".pf-c-modal-box__footer button:contains(Add)")
+
+                if (self.xfail):
+                    # Check incomplete dialog
+                    b.wait_in_text(".pf-c-modal-box .pf-c-alert.pf-m-danger", self.xfail)
+                    self.cancel()
+                else:
+                    b.wait_not_present("#add-new-static-entry")
+
+            def verify_dialog(self):
+                # Check that the defined network is now visible
+                b.wait_in_text("body", "Networks")
+
+                # Verify libvirt XML
+                net_xml = m.execute(f"virsh -c qemu:///system net-dumpxml {self.network_name}")
+                network_node = ET.fromstring(net_xml)
+                host_by_mac = network_node.find(f'./ip/dhcp/host[@mac="{self.mac}"]')
+                if host_by_mac is None:
+                    raise Exception(f'No DHCP host with MAC address {self.mac} found in network XML:\n{net_xml}')
```

Comment:
A cleaner way to write this would be `self.assertTrue(host_by_mac, f'...error')`. This would also avoid raising the overly generic Exception base class.